xarray/backends/netCDF4_.py

```diff
@@ -242,13 +243,21 @@ def _extract_nc4_variable_encoding(
         "shuffle",
         "_FillValue",
         "dtype",
+        "compression",
     }
     if lsd_okay:
         valid_encodings.add("least_significant_digit")
     if h5py_okay:
-        valid_encodings.add("compression")
         valid_encodings.add("compression_opts")
 
+    if backend == "netCDF4" and has_netcdf4:
+        # if using netCDF4 check the versions are updated enough for supporting the new
+        # compression
+        if Version(netCDF4.__version__) < Version("1.6.0") or Version(
+            netCDF4.getlibversion().split(" ")[0]
+        ) < Version("4.9.0"):
+            valid_encodings.remove("compression")
```

Comment:
IMO we can just let netCDF raise the error for this case. 

We should pass along `compression` if it exists but not add or remove it by default.